tests/contrib/fastapi/test_fastapi.py

```diff
@@ -589,3 +603,20 @@ def test_host_header(client, tracer, test_spans, host):
     assert test_spans.spans
     request_span = test_spans.spans[0]
     assert request_span.get_tag("http.url") == "http://%s/asynctask" % (host,)
+
+
+def test_tracing_in_middleware(app_with_middleware, test_spans):
```

Comment:
Would a snapshot test work better here?

Basically we want to assert that the normal `fastapi.request` span exists along with the customer user middleware span?